### PR TITLE
Add Dispatchfile

### PR DESCRIPTION
**Pull request by AliakseiKhomchanka**
Adding Dispatch integration for this repo.

### CODE REVIEW

**Comment by shaneutt** [comment]
Looks like there's some problems in CI @AliakseiKhomchanka, I've added the other codeowners here let us know if you require any assistance.

File: Dispatchfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+              image = "mesosphere/kubeaddons-ci:latest",
+              command = ["/bin/bash", "-c"],
+              args = [
+                  """
+                  set -euo pipefail
+                  export GIT_TERMINAL_PROMPT=1
+                  export GOPRIVATE=github.com/mesosphere/kubeaddons
+                  ssh-keyscan -t rsa github.com > /etc/ssh/known_hosts
+                  git config --global url."https://$GITHUB_TOKEN:@github.com/".insteadOf "https://github.com/"
+                  make kind-test
```
Comment (kensipe) [inline Dispatchfile:30]: do we want to spin up a kind cluster?
does the underlying node support the creation of kind?

```
 Cannot connect to the Docker daemon at unix:///var/run/docker.sock. 
```

Comment (shaneutt) [reply Dispatchfile:30]: More recent kind should be playing nicely with podman too, if docker ends up being an issue for some reason? we might have luck there

Comment (AliakseiKhomchanka) [reply Dispatchfile:30]: It's supposed to be using cluster claim controller to get kubeconfig of a kind cluster that is already running somewhere (from what I understand about cluster claim controller) so I'm not entirely sure why the container needs to be privileged.

**Comment by shaneutt** [comment]
@AliakseiKhomchanka looks like we're getting close here, need any help from @mesosphere/sig-ksphere-catalog to get this one over the finish line?

**Comment by AliakseiKhomchanka** [comment]
@shaneutt The test seems to be broken, Dispatch itself is alright, I think. I'm not a specialist regarding what those tests actually do, anyone that knows?

**Comment by shaneutt** [comment]
> @shaneutt The test seems to be broken, Dispatch itself is alright, I think. I'm not a specialist regarding what those tests actually do, anyone that knows?

I was under the impression that the `integration-test` is identical to the teamcity CI, so if TC is passing and dispatch is not, I would expect we have a problem that may be dispatch specific?

Perhaps @faiq is a good resource

**Comment by shaneutt** [comment]
Please see context from https://github.com/mesosphere/kubeaddons-enterprise/pull/75#pullrequestreview-480829860 and the JIRA we've created to follow up: https://jira.d2iq.com/browse/D2IQ-71485

Dispatch is now disabled in this repo with the above JIRA being the follow up and this PR is now linked to that JIRA as a PR to re-open once we re-enable it. I'm gonna close this for now so we can prioritize it later as that seems to make sense based on the current context, however if you feel I'm wrong to do so and there's more work to be done immediately in this PR please don't hesitate to re-open this and scold me :)

**Comment by AliakseiKhomchanka** [comment]
Everything seems to be working, need a review and can merge.

**Review by faiq — DISMISSED** [review]

**Comment by faiq** [comment]
Squash commits

File: Dispatchfile [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+                  make dispatch-test
+                  """
+              ],
+              resources = k8s.corev1.ResourceRequirements(
+              limits = {
+                "cpu": k8s.resource_quantity("8000m"),
+                "memory": k8s.resource_quantity("6Gi")
+              },
+              requests = {
+                "cpu": k8s.resource_quantity("1000m"),
```
Comment (faiq) [inline Dispatchfile:43]: You can probably request a lot less than this maybe 500m

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
-KUTTL_VERSION=0.5.0
+KUTTL_VERSION=0.6.1
```
Comment (faiq) [inline Makefile:1]: Why are you bumping kuttl?

Comment (AliakseiKhomchanka) [reply Makefile:1]: That's the version that the current build config in TeamCity uses, just making sure Dispatch one is up to date too.

File: Dispatchfile (hunk 2 above)
Comment (AliakseiKhomchanka) [reply Dispatchfile]: Updated.

**Review by faiq — APPROVED** [review]

**Review by kensipe — APPROVED** [review]

**Review by alejandroEsc — DISMISSED** [review]
Lgtm

**Comment by makkes** [comment]
It looks like the Jenkins tests do fail on every PR in this repo right now.

**Comment by rishabh96b** [comment]
@makkes and everyone, Jenkins test is fixed by https://github.com/mesosphere/kubeaddons-enterprise/pull/98 . We should be good to go for this PR now.

**Review by faiq — CHANGES_REQUESTED** [review]
the make target should be the same for running locally as running on dispatch

File: Makefile [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 KIND_VERSION=0.8.1
 KUBERNETES_VERSION ?= 1.17.5
-KUBECONFIG=kubeconfig
+KUBECONFIG?="kubeconfig"
```
Comment (joejulian) [inline Makefile:4]: nit: convention
```suggestion
KUBECONFIG ?= kubeconfig
```

File: Makefile [hunk 5]
```diff
@@ -27,3 +27,13 @@ ifneq (,$(wildcard kubeaddons-tests/Makefile))
...
+
+.PHONY: dispatch-test
+dispatch-test: 
+	mkdir -p bin/
+	curl -Lo bin/kubectl-kuttl_$(KUTTL_VERSION) https://github.com/kudobuilder/kuttl/releases/download/v$(KUTTL_VERSION)/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)_$(MACHINE)
+	chmod +x bin/kubectl-kuttl_$(KUTTL_VERSION)
+	ln -sf ./kubectl-kuttl_$(KUTTL_VERSION) bin/kubectl-kuttl
+	KUBEADDONS_TESTS_KUBECONFIG=/workspace/src-git/kubeconfig
+	git clone https://github.com/mesosphere/kubeaddons-tests.git --branch master --single-branch
+	kubeaddons-tests/run-tests.sh
```
Comment (joejulian) [inline Makefile:39]: Can you take another stab at this? It's preferable that a user run the exact same make command as any CI to run tests. Do you think there might be a way to do that?

**Review by joejulian — CHANGES_REQUESTED** [review]

**Comment by makkes** [comment]
This PR has been superseded by #109.

**Pull request closed without merging**